common/src/main/scala/explore/common/ObsQueries.scala

```diff
@@ -34,9 +35,24 @@ object ObsQueries {
         Pointing.PointingAsterism(id, name, Nil)
     }
 
+  implicit class ConstraintSetsObsQueryDataOps(
+    val self: ProgramObservationsQuery.Data.ConstraintSets.Nodes
+  ) extends AnyVal {
+    def asConstraintSummary: ConstraintsSummary =
+      new ConstraintsSummary {
+        val id              = self.id
+        val name            = self.name
+        val imageQuality    = self.imageQuality
+        val cloudExtinction = self.cloudExtinction
+        val skyBackground   = self.skyBackground
+        val waterVapor      = self.waterVapor
+      }
+  }
+
```

Comment:
This can be avoided by making `ProgramObservationsQuery.Data.ConstraintSets.Nodes` extend `ConstraintsSummary`. Clue's generator can be instructed to do so by declaring `trait Nodes extends ConstraintsSummary` in the object hierarchy.

For example: https://github.com/gemini-hlsw/explore/blob/master/common-graphql/src/main/scala/explore/common/ObsQueriesGQL.scala#L48

Reply:
Ok I'll try that